### PR TITLE
1337x Search Plugin

### PR DESCRIPTION
**Pull request by OmegaZero**
### Motivation for changes:
Add 1337x search plugin

### Detailed changes:
- Adds UrlRewriter and search plugin for search 1337x.to

### Config usage if relevant (new plugin or updated schema):
```Either
    - 1337x: yes
or
    - 1337x:
         order_by: (seeders|leechers|time|size)

```
### Log and/or tests output (preferably both):

(removed due copyrighted material)

### CODE REVIEW

File: flexget/plugins/search/1337x.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            if config['order_by'] != 'leechers':
+                order_by = '/{0}/desc'.format(config['order_by'])
+                sort_order = 'sort-'
+
+        entries = set()
+
+        for search_string in entry.get('search_strings', [entry['title']]):
+
+            query = '{0}search/{1}{2}/1/'.format(sort_order,quote(search_string.encode('utf8')), order_by)
+            log.debug('Using search params: {0}; ordering by: {1}'.format(search_string, order_by or 'default'))
```
Comment (cvium) [inline flexget/plugins/search/1337x.py]: I didn't notice this one before. When logging you should let the logger handle the string formatting eg. `log.debug('some logging %s with multiple vars %s', var1, var2)`

**Comment by cvium** [comment]
Looks like you have some leftover garbage. You should remove the dir `flexget/plugins/daemon/irc_bot`. Not related to your PR but it's something that was removed from Flexget a while ago.

**Comment by OmegaZero** [comment]
yeah I never had that issue until I got the next branch today...

File: flexget/plugins/search/1337x.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+from builtins import *  # pylint: disable=unused-import, redefined-builtin
+
+import logging
+import re
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.requests import TimedLimiter, RequestException
+from flexget.utils.requests import Session as RequestSession
```
Comment (cvium) [inline flexget/plugins/search/1337x.py]: You should remove any unused imports.

File: flexget/plugins/search/1337x.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+from flexget.utils.soup import get_soup
+from flexget.utils.search import torrent_availability
+from flexget.plugins.plugin_urlrewriting import UrlRewritingError
+
+log = logging.getLogger('1337x')
+
+requests = RequestSession()
```
Comment (cvium) [inline flexget/plugins/search/1337x.py]: I thought you removed this?

File: flexget/plugins/search/1337x.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+from future.moves.urllib.parse import quote
+from builtins import *  # pylint: disable=unused-import, redefined-builtin
+
+import logging
+import re
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.requests import TimedLimiter, RequestException
```
Comment (cvium) [inline flexget/plugins/search/1337x.py]: Please go through all your imports and make sure they are used. You are not using TimedLimiter.

**Pull request merged**